generator/mavgen.py

```diff
@@ -34,23 +40,45 @@ def mavgen(opts, args) :
     # Enable validation by default, disabling it if explicitly requested
     if opts.validate:
         try:
-            from lib.genxmlif import GenXmlIfError
-            from lib.minixsv import pyxsval
+            from lxml import etree
+            with open(schemaFile, 'r') as f:
+                xmlschema_root = etree.parse(f)
+                xmlschema = etree.XMLSchema(xmlschema_root)
         except:
-            print("WARNING: Unable to load XML validator libraries. XML validation will not be performed")
+            print("WARNING: Unable to load XML validator libraries. XML validation will not be performed", file=sys.stderr)
             opts.validate = False
 
-    def mavgen_validate(fname, schema, errorLimitNumber) :
-        """Uses minixsv to validate an XML file with a given XSD schema file. We define mavgen_validate
+    def mavgen_validate(xmlfile):
+        """Uses lxml to validate an XML file. We define mavgen_validate
            here because it relies on the XML libs that were loaded in mavgen(), so it can't be called standalone"""
-        # use default values of minixsv, location of the schema file must be specified in the XML file
-        domTreeWrapper = pyxsval.parseAndValidate(fname, xsdFile=schema, errorLimit=errorLimitNumber)
+        xmlvalid = True
+        try:
+            with open(xmlfile, 'r') as f:
+                xmldocument = etree.parse(f)
+                xmlschema.assertValid(xmldocument)
+                forbidden_names_re = re.compile("^(break$|case$|class$|catch$|const$|continue$|debugger$|default$|delete$|do$|else$|\
+                                    export$|extends$|finally$|for$|function$|if$|import$|in$|instanceof$|let$|new$|\
+                                    return$|super$|switch$|this$|throw$|try$|typeof$|var$|void$|while$|with$|yield$|\
+                                    enum$|await$|implements$|package$|protected$|static$|interface$|private$|public$|\
+                                    abstract$|boolean$|byte$|char$|double$|final$|float$|goto$|int$|long$|native$|\
+                                    short$|synchronized$|transient$|volatile$).*", re.IGNORECASE)
+                for element in xmldocument.iter("*"):
+                    if element.tag in ["enum", "entry", "message", "field"]:
+                        match = forbidden_names_re.search(element.get('name'))
+                        if match is not None:
+                            print("Validation error:", file=sys.stderr)
+                            print("Element : %s at line : %s contains forbidden word" % (element.tag, element.sourceline), file=sys.stderr)
+                            xmlvalid = False
```

Comment:
Check this comment: https://github.com/ArduPilot/pymavlink/pull/3#issuecomment-250541991